### PR TITLE
Preconnect processing via fetch

### LINKED ISSUE #34: Explain preconnect in terms of Fetch
There's a disconnect right now between how a rel="preconnect" behaves and how that aligns to the equivalent fetch operation from https://fetch.spec.whatwg.org 

That is, preconnect exists to 'warm up' the connection used as part of the [HTTP network fetch](https://fetch.spec.whatwg.org/#http-network-fetch) phase of Fetch, specifically step 1. The inputs for such an operation are the optional _credentials flag_.

The _credentials flag_ is an internal implementation detail of Fetch (more clearly noted in [HTTP network or Cache Fetch](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch) ), whose value is computed during the [HTTP fetch](https://fetch.spec.whatwg.org/#http-fetch) stage of Fetch

The following text from Fetch explains the conversion

> Let _credentials_ flag be set if either request's credentials mode is "`include`", or request's
> credentials mode is "`same-origin`" and the _CORS flag_ is unset, and unset otherwise.

There's a disconnect in the HTML spec's [description of the _crossorigin_ attribute](https://html.spec.whatwg.org/#cors-settings-attribute) and how that affects fetch, mostly because HTML isn't fully referring to Fetch, but the concept is hopefully clear - "anonymous" is equivalent to setting _omit credentials flag_ (HTML terms), which is most roughly equivalent to setting the Request's  [_credentials mode_](https://fetch.spec.whatwg.org/#concept-request-credentials-mode) to `omit` (Fetch terms).

The reason I raise this is to ensure that when setting a _crossorigin_ attribute on a rel="preconnect", the concepts can be clear as to how they align with Fetch's HTTP Network Fetch phase 1, and in particular, the distinct pools that exist that are exclusively keyed off the Fetch _credentials flag_.

Explaining in terms of HTML Standard, as done currently, leaves ambiguity, because there isn't a described Request object as there is in Fetch, and rel="preconnect" isn't described in terms of HTML's (outdated) [CORS-enabled fetch](https://html.spec.whatwg.org/#potentially-cors-enabled-fetch). As a result, there's an impedance and terminology mismatch.

### PR DESCRIPTION
**Pull request by igrigorik**
Closes #34.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -96,11 +96,16 @@
   &lt;link rel="preconnect" href="//cdn.example.com" crossorigin&gt;
       </pre>
 
-      <ul>
-        <li>The `crossorigin` <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#cors-settings-attributes">CORS setting attribute</a> is an OPTIONAL attribute that indicates the CORS policy for resources that will be fetched from the specified origin - e.g. when set to "anonymous", the user agent SHOULD initiate a handshake for a connection that can be used to fetch resources with an equivalent policy.</li>
+      <p>To <dfn>initiate a preconnect</dfn>, the user agent MUST run these steps:</p>
+      <ol>
+        <li>Let <var>origin</var> be the origin [[!ORIGIN]] of the value of the `href` attribute.</li>
```
Comment (annevk) [inline index.html]: Don't you need to parse the value first?

File: index.html [hunk 2]
```diff
@@ -96,11 +96,16 @@
...
+        <li>If the `crossorigin` <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#cors-settings-attributes">CORS setting attribute</a> value is set to `use-credentials` and <var>origin</var> is not equal to <a href="https://html.spec.whatwg.org/multipage/browsers.html#origin-2">current Document's origin</a>, let <var>credentials</var> be a boolean value set to `true`. Otherwise, let <var>credentials</var> be a boolean value set to `false`.</li>
```
Comment (annevk) [inline index.html]: I think it would be better to use the terminology from HTML here. That the attribute is in a certain state.

Comment (annevk) [inline index.html]: Also, `anonymous` is the value you need to pair the origin comparison with, not `use-credentials`...

File: index.html (hunk 1 above)
Comment (annevk) [inline index.html]: Also, you should probably check the scheme is either `http` or `https`.

Comment (igrigorik) [inline index.html]: Why http vs https? Wouldn't that be captured by 'origin'?

**Comment by igrigorik** [comment]
@annevk updated, getting warmer? :)

**Comment by annevk** [comment]
This looks reasonable. What about the http/https scheme check?

**Comment by igrigorik** [comment]
Hmm, I left a comment about that but I guess github ate it once I pushed the update.. Why do we need http/https check here? Shouldn't that already be taken care of via "origin" comparisons?

**Comment by igrigorik** [comment]
@sleevi can you take a quick pass over this as well? :)

**Comment by annevk** [comment]
@igrigorik I was thinking the http/https check could be good since nothing else will really work. In Fetch you can only get there anyway if the scheme is either http/https.

**Comment by igrigorik** [comment]
Oh, are you saying that we should check for http/https scheme on the parsed URL - i.e. exclude file://, etc? I was interpreting your comment as 'we need to distinguish http from https'.

**Comment by annevk** [comment]
I was trying to say the former.

**Comment by igrigorik** [comment]
@annevk updated, ptal.

**Comment by igrigorik** [comment]
@annevk ack, fixed.

**Pull request merged**